docs/js/features/odata/use-generated-client.md

```diff
@@ -0,0 +1,390 @@
+
+On an abstract level, creating a request using the fluent api always follows the same simple structure:
+
+```ts
+return MyEntity.requestBuilder()
+    .<requestType>(...)
+    .<additionalRequestConfiguration>(...)
+    .execute(destination);
+```
+
+Every entity has a `requestBuilder` function, that allows to chain all types of request builders that are available for this entity, e. g. `MyEntity.requestBuilder().getAll()` for the [`getAll`](#GetAll) request type.
+
+Potential request types (denoted by `requestType` in the example above) are [`create`](#Create), [`getAll`](#GetAll), [`getByKey`](#getByKey), [`update`](#Update) and [`delete`](#Delete).
+
+:::note
+Some entities do not support all of the request types, which in turn won't be available through the API.
+:::
+
+The request can further be configured by chaining additional configuration functions (denoted by `additionalRequestConfiguration` in the example above). All requests can be configured by setting [custom request headers](#Setting-custom-request-headers), [custom query parameters](#Setting-custom-query-parameters) and a [custom service path](#Setting-custom-service-path).
+
+Each request type has additional request specific configuration options:
+
+Creating  an entity `asChildOf` another entity for [`create`](#Create) requests, [ETag handling](#Handling-of-ETags) for [`update`](#Update) and [`delete`](#Delete) requests, as well as set operations for [`getAll`](#GetAll) requests and [`select`](#Select)ing properties for [`getAll`](#GetAll) and [`getByKey`](#GetByKey) requests.
+<!--
+*3 if there was a specific section on create this should rather be explained there and in more detail.
+-->
+
+The last step when making a request using the SAP Cloud SDK is the request execution. Once the request is configured, chain the `execute` function and pass a [destination](/cloud-sdk/docs/js/features/connectivity/destination-js-sdk/) to it.
+
+### Setting custom request headers
+
+To use the SAP API Business Hub sandbox for your requests, you will need to pass the `apikey` using the `withCustomHeaders` method. You can also set the `authorization` with this method.
+
+:::caution
+Setting `authorization` (and `apikey`) headers will overwrite any automatic authorization header building that the SDK would normally do.
+:::
+
+Checkout the following example:
+
+```ts
+MyEntity.requestBuilder()
+    .getAll()
+    .withCustomHeaders({
+        apikey: 'my-api-key'
+    });
+```
+
+### Setting custom query parameters
+
+This method allows you to set key-value pairs in the request
+Key-value pairs denoting additional custom query parameters to be set in the request
+
+:::caution
+Setting custom query parameters overwrites any other configuration that might have been created by the SDK.
+:::
+
+```ts
+MyEntity.requestBuilder()
+    .getAll()
+    .withCustomQueryParameters({
+        language: 'en'
+    });
```

Comment:
```suggestion
```suggestion
The SAP Cloud SDK automatically sets some necessary request headers on every request. You can specify additional custom headers using the `withCustomHeaders` function:

```ts
MyEntity.requestBuilder()
    .getAll()
    .withCustomHeaders({
        apikey: 'my-api-key'
    });

```